pkg/controller/handlers/task/task.go

```diff
@@ -0,0 +1,373 @@
+)
+
+var ErrorValidating = errors.New("found existing external interface, only one external interface can be set")
+
+type Handler struct {
+}
+
+func NewHandler() *Handler {
+	return &Handler{}
+}
+
+func (t *Handler) HandleTaskCreationForCapabilities(req router.Request, _ router.Response) error {
+	thread := req.Object.(*v1.Thread)
+
+	if !thread.Spec.Project {
+		return nil
+	}
+
+	if err := Validate(thread, false); err != nil {
+		return err
+	}
+
+	if err := t.UpdateTaskForSlack(thread, req); err != nil {
+		return err
+	}
+
+	if err := t.UpdateTaskForDiscord(thread, req); err != nil {
+		return err
+	}
+
+	if err := t.UpdateTaskForEmail(thread, req); err != nil {
+		return err
+	}
+
+	if err := t.UpdateTaskForWebhook(thread, req); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func Validate(thread *v1.Thread, forSlack bool) error {
+	if forSlack {
+		if thread.Spec.Capabilities.OnDiscordMessage || thread.Spec.Capabilities.OnEmail != nil || thread.Spec.Capabilities.OnWebhook != nil {
+			return ErrorValidating
+		}
+	}
+	var count int
+	if thread.Spec.Capabilities.OnSlackMessage {
+		count++
+	}
+	if thread.Spec.Capabilities.OnDiscordMessage {
+		count++
+	}
+	if thread.Spec.Capabilities.OnEmail != nil {
+		if forSlack {
+			count++
+		}
+		count++
+	}
+	if thread.Spec.Capabilities.OnWebhook != nil {
+		count++
+	}
+	if count > 1 {
+		return ErrorValidating
+	}
+	return nil
+}
+
+func (t *Handler) UpdateTaskForSlack(thread *v1.Thread, req router.Request) error {
+	workflow := v1.Workflow{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name.SafeHashConcatName(system.WorkflowPrefix, "slack", thread.Name),
+			Namespace: thread.Namespace,
+		},
+		Spec: v1.WorkflowSpec{
+			ThreadName: thread.Name,
+			Manifest: types.WorkflowManifest{
+				OnSlackMessage: &types.TaskOnSlackMessage{},
+				Steps: []types.Step{
+					{
+						Step: "Reply back to the user with a message in the thread",
+					},
+				},
+				Name:        "Slack Integration Task",
+				Description: "This task is used to integrate with Slack.",
+			},
+		},
+	}
+
+	if thread.Spec.Capabilities.OnSlackMessage {
+		if err := req.Get(&workflow, workflow.Namespace, workflow.Name); kclient.IgnoreNotFound(err) != nil {
+			return err
+		} else if err != nil {
+			alias, err := randomtoken.Generate()
+			if err != nil {
+				return err
+			}
+			if len(alias) > 12 {
+				alias = alias[:12]
+			}
+			workflow.Spec.Manifest.Alias = alias
+			if err := req.Client.Create(req.Ctx, &workflow); err != nil {
+				return err
+			}
+			thread.Status.WorkflowNameFromIntegration = workflow.Name
+		}
+	} else {
+		if err := req.Get(&workflow, workflow.Namespace, workflow.Name); kclient.IgnoreNotFound(err) != nil {
+			return err
+		} else if err == nil {
+			if err := req.Client.Delete(req.Ctx, &workflow); err != nil && !apierrors.IsNotFound(err) {
+				return err
+			}
+			if thread.Status.WorkflowNameFromIntegration == workflow.Name {
+				thread.Status.WorkflowNameFromIntegration = ""
+			}
```

Comment:
No harm in just setting it, here and similar places.
```suggestion
			thread.Status.WorkflowNameFromIntegration = ""
```